catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -232,6 +233,90 @@ private Filter transformCustomFunctionToFilter(Filter filter) {
         return filter;
     }
 
+    private QueryRequest normalizeSort(QueryRequest request) {
+        if (request == null) {
+            return null;
+        }
+
+        Query query = request.getQuery();
+        if (query == null) {
+            return request;
+        }
+
+        List<SortBy> sortBys = new ArrayList<>();
+
+        // Primary sort parameter
+        SortBy sortBy = query.getSortBy();
+        SortBy newSortBy = normalizeSortBy(sortBy);
+        if (newSortBy != null) {
+            sortBys.add(newSortBy);
+        }
+
+        // Additional sort parameters
+        Serializable extraSortBys = request.getPropertyValue(EXT_SORT_BY);
+        if (extraSortBys != null && extraSortBys instanceof SortBy[]) {
+            List<SortBy> extraSortBysList = Arrays.asList((SortBy[]) extraSortBys);
+            extraSortBysList.stream()
+                    .map(this::normalizeSortBy)
+                    .filter(Objects::nonNull)
+                    .forEach(sortBys::add);
+        }
+
+        Map<String, Serializable> newProperties = request.getProperties();
+        SortBy normalizedSortBy = null;
+        if (sortBys.size() > 0) {
+            normalizedSortBy = sortBys.get(0);
+        }
+        if (sortBys.size() > 1) {
+            SortBy[] extraSortBysArray = sortBys.subList(1, sortBys.size())
```

Comment:
[pencil] If you accept my proposal on line 257 to make the `SortBy[]` instance check an assertion, then you can cast `extraSortBys` to a `SortBy[]`, then you can replace this tricky `sortBys.subList(...).toArray(...)` with `extraSortBys`.
@ line 256:
```
Serializable extraSortBysRaw = request.getPropertyValue(EXT_SORT_BY);
assert(extraSortBysRaw, instanceOf(SortBy[].class));
SortBy[] extraSortBys = (SortBy[]) extraSortBysRaw;
if (extraSortBys != null) {...
```
@ line 270:
```
if (sortBys.size() > 1) {
    newProperties.put(EXT_SORT_BY, extraSortBys);
}
```

Reply:
This logic isn't just used to cast `extraSortBys` to an array though. It's used to strip off the leading element.

Reply:
I forgot that the primary `sortBy` might not be added, so that sublist is not necessarily equivalent to `extraSortBys`. My bad.